irohad/pending_txs_storage/impl/pending_txs_storage_impl.cpp

```diff
@@ -64,10 +68,29 @@ PendingTransactionStorageImpl::getPendingTransactions(
   while (account_batches.batches.end() != batch_iterator
          and remaining_space >= batch_iterator->get()->transactions().size()) {
     auto &txs = batch_iterator->get()->transactions();
-    response.transactions.insert(
-        response.transactions.end(), txs.begin(), txs.end());
-    remaining_space -= txs.size();
-    ++batch_iterator;
+    // can be easily extended by adding more conditions if needed
+    if (first_tx_time or last_tx_time) {
+      auto time_predicate = [&](auto &tx) {
+        bool first_tx_time_valid =
+            first_tx_time ? first_tx_time < tx->createdTime() : true;
+        bool last_tx_time_valid =
+            last_tx_time ? tx->createdTime() < last_tx_time : true;
+        return first_tx_time_valid && last_tx_time_valid;
+      };
+      auto filtered_txs_size = std::count_if(txs.begin(), txs.end(), time_predicate);
+      std::copy_if(txs.begin(),
+                   txs.end(),
+                   std::back_inserter(response.transactions),
+                   time_predicate);
+      remaining_space -= filtered_txs_size;
```

Comment:
I believe it could be changed to 
```suggestion
      remaining_space -= response.transactions.size();
```
and do not count_if

Reply:
this will work of response.transactions is not empty
```
auto iter_before_copy = std::end(response.transactions);
auto iter_after_copy = copy_if(........);
remaining_space -= std::distance(iter_before_copy,iter_after_copy);
```

Reply:
Well, I think it would work when response.transactions is empty. `transactions` is part of `PendingTransactionStorage::Response` and will be initialized as default vector